services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +43,41 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
+  const void * itemToQueue = (const void *)event;
+  BaseType_t sendEventResult = xQueueSend(thermalMgrQueueHandle, itemToQueue, 0);
+  if (sendEventResult != pdTRUE) return ERR_CODE_QUEUE_FULL;
 
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t checkOverUnder;
+  checkOverUnder.type = THERMAL_MGR_EVENT_OVER_UNDER;
+
+  /* Add interrupt event to the queue */
+  thermalMgrSendEvent(&checkOverUnder);
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  lm75bd_config_t config = *(lm75bd_config_t *)pvParameters;
+  float currTemp;
+  thermal_mgr_event_t nextEvent;
```

Comment:
in general try to limit the scope of variables as small as possible so these should be declared in the while loop